lib/modules/datasource/rubygems/get.ts

```diff
@@ -182,4 +200,87 @@ export class InternalRubyGemsDatasource extends Datasource {
 
     return new Marshal(response.body).parsed as T;
   }
+
+  private async isNexusDataSource(registry: string): Promise<boolean> {
+    const statusEndPoint = '/service/rest/v1/status';
+    const endPointRe = regEx(`^(?<endPoint>.*:\\d{1,5})\\/`);
+    const { endPoint } = registry.match(endPointRe)?.groups ?? {
+      endPoint: '',
+    };
+    const nexusUrl = endPoint.concat(statusEndPoint);
+    let isNexus = false;
+    try {
+      const response = await this.http.getJson<HttpResponse>(nexusUrl);
+      isNexus = response.headers?.server?.includes('Nexus') ?? false;
+    } catch (err) {
+      logger.debug({ err }, 'Not a nexus datasource');
```

Comment:
If we are dealing with a non-Nexus registry, will this log `err` to debug logs once per dependency?

Reply:
This is perhaps a bigger problem than the inefficient http requests. Users will get pages of unnecessary error logs

Reply:
it's OK to keep it as long as you have only 1 call to the nexus http , so this log wouldn't be printed for Every dependency that comes into here.
this can be resolved if you do 1 call to the nexus and cache the results.

Reply:
There is no point logging the `err` here. It is not an error situation, and the trace it will log is of no use to someone who just happens to run one of the many non-Nexus private gem servers.

Reply:
@rarkins @PhilipAbed Note that this log is only present in the fall back flow. like this log here https://github.com/renovatebot/renovate/blob/b37c90650af3dffe7097e7b01e886cb7c9382034/lib/modules/datasource/rubygems/get.ts#L68
We need to catch for errors incase when the request fails to endpoint. I've changed it to `logger.debug('${nexusEndPoint} is not a nexus datasource');`

Open for better suggestions